src/libraries/System.Numerics.Tensors/src/System/NIndex.cs

```diff
@@ -14,6 +14,7 @@ namespace System.Buffers
     /// int lastElement = someArray[^1]; // lastElement = 5
     /// </code>
     /// </remarks>
+    [Experimental("SNTEXP0001")]
```

Comment:
Can / should we follow the approach done for obsoletions, with a main list of all the values and strongly-typed references to them? e.g.
https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/System/Obsoletions.cs
https://github.com/dotnet/runtime/blob/2746c037b32a940341dbff73acd335215cfcf90a/src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs#L1032

Reply:
That seems like a good idea. We'll want to avoid recycling the IDs, so we need to keep the historical values around somewhere. Having the analogous `Experimentals.cs` and then adding experimental IDs to [docs/project/list-of-diagnostics.md](https://github.com/dotnet/runtime/blob/main/docs/project/list-of-diagnostics.md) would set us up for better maintainability therein.

@tannergooding -- I'll push a commit to the PR that incorporates that.